### ISSUE TITLE
Any plans on porting the lastest version to .NET?

### ISSUE THREAD

**Issue by jasmh** [KosherJava/zmanim#167]
Thanks.

**Comment by BehindTheMath** [comment]
There's already a .NET port:

https://github.com/Yitzchok/Zmanim

**Comment by jasmh** [comment]
Yes, but the last version (1.5) vas ported 3 years ago and it's missing a lot that's been added since then.

**Comment by BehindTheMath** [comment]
Feel free to open an issue on that repo. I don't maintain it.

**Comment by jasmh** [comment]
I did, I thought you might have been connected somehow.

Thanks anyway.

**Comment by KosherJava** [comment]
@chacham2 and @jasmh  There was recently some activity in the .NET port, and the author is hopeful to update it. They would gladly accept any PRs that would start bringing it up to date.

**Comment by chacham2** [comment]
@KosherJava That project is neither complete nor up-to-date, so even if the port was finished, it would still likely lag behind this one. This afternoon, i was thinking that ikvm might be the best solution, and certainly ought to be the easiest.

That being said, ikvmc compiles the jar without issue, but it seems .net core is not working well, as evidenced by some reported issues (https://github.com/ikvm-revived/ikvm/issues). For the runtime i used the Windward version (https://www.nuget.org/packages/IKVM.WINDWARD) on NuGet. Tomorrow, i will likely try Framework instead.

**Comment by Yitzchok** [comment]
@chacham2 Not complete as in? Yes, it is not up-to-date and I can use the community help to get it there.

ikvm is not a clean solution to use java code in .NET you end up using way to many java namespaces in order to work with the java code which isn't optimal.

**Comment by chacham2** [comment]
@Yitzchok I apologize: incomplete was an assumption based on the date and low usage.

>it is not up-to-date and I can use the community help to get it there.

I will be frank and say upfront that i am not interested in helping with anything other than kudos. I applaud the effort, in any case.

While i am not qualified to comment on it overall, my own little mindset for my own little project is that the tradeoff will probably be worth the benefit of having the latest version of an actively maintained project. Then again, as i got nowhere today with my foray into unfamiliar territory, i might just be forced to revise that ill-formed opinion.

**Comment by chacham2** [comment]
In case anyone else is interested, it works fine in ikvm, using framework. Here is a simple example in vb winforms, where Name, Latitude, and Longitude are constants specified in another file.

```
Imports com.kosherjava.zmanim

Public Class Form1
    Private Sub Form1_Load(Sender As Object, Arguments As EventArgs) Handles MyBase.Load

        Dim Day As Date
        Dim Zmanim As New ZmanimCalendar(New util.GeoLocation(Name, Latitude, Longitude, java.util.TimeZone.getTimeZone(TimeZone)))

        For Offset As Integer = 0 To 3
            Day = Now.AddDays(Offset)
            Zmanim.getCalendar.set(Day.Year, Day.Month, Day.Day)
            Debug.WriteLine(Zmanim.getMinchaGedola)
            'Debug.WriteLine(Zmanim.toString())
        Next

        Close()
    End Sub
End Class
```

**Comment by KosherJava** [comment]
If anyone volunteers to bring .NET up to date, please let me know and I will update [Ports to Other Languages](https://github.com/KosherJava/zmanim#ports-to-other-languages) section. This is not an issue with the current Java based project, and I am closing it out.
